### PR TITLE
resample v2

### PR DESCRIPTION
**Pull request by buthanoid**

### CODE REVIEW

**Comment by buthanoid** [comment]
function `addFitsImageHDU` presents a bug: the result HDU is not added to the image library.
Scenario: start with new. load a oifitsfile and some init image. select WISARD and an init image that we will call "PI_GRU_32x32_50". Click on Run. Observe that the result HDU has not been added to the library. Indeed it also have the name "PI_GRU_32x32_50" that is why it has been rejected.
Previous version `addFitsImageHDUs` was different because it was not rejecting every HDU: as soon as one would have been rejected it would break from the loop, why ?
Both versions seems buggy from my point of view. It should not reject a HDU because it has same name that the one in input form. How does it know if it is different HDU or same ? I think we should remove this test. There is a second test which test reference equality and checksum equality which seems more useful. There is also another procedure after it to avoid name conflict.

**Comment by bourgesl** [comment]
Let's fix that bug in live now ?

**Comment by bourgesl** [comment]
This test is a bad solution to filter out the INIT_IMG and RGL_PRIOR image coming from service results ... in order to avoid adding them (repeatedly) in the image library.
Warning: checksums are often different (murphy's law) between the HDU (memory) and the HDU coming from IR software (header rewriting ...)

**Comment by bourgesl** [comment]
I pushed a workaround, could you test again your case "PI_GRU_32x32_50" ?
See https://github.com/JMMC-OpenDev/oimaging/pull/35/commits/a7d34fc977fa24a16446862e046cbdcdcae34b3f
I noticed that checksum check is helpful to skip results that are exactly identical to previous ones: it happens once BSMEM or any IR software converge : same input => same result (repeatably)

**Comment by buthanoid** [comment]
Still the issue:
- adding PI_GRU_32x32_50 to library
- selecting it as INIT_IMG
- Run WISARD
- Obtain a Service Result with two HDUs: PI_GRU_32x32_50 and PI_GRU_32x32_50
- Get into the function addServiceResult()
- It calls function addFitsImageHDUs() with third param = getInputImageRefs(serviceResult.getOifitsFile()), which is equals to Set(PI_GRU_32x32_50)
- Computation of checkSelectedImageHDUs which is equals to true
- Call of function addFitsImageHDU(FitsImageHDU, String, boolean)
- failing test and displaying: "skipping image hdu 'PI_GRU_32x32_50' : already present (InputImage)" twice: once for INIT_IMG and once for RLG_PRIO

**Comment by bourgesl** [comment]
Your case is really not representative as your HDU_NAME is in fact coming from a previous WISARD file !
We are reaching the limits: ''PI_GRU_32x32_50' as input, gives you two HDUs: PI_GRU_32x32_50 and PI_GRU_32x32_50 !
The whole chain (GUI + wisard) is then corrupted as the ImageOI file should never contain 2 HDUs with the same HDU_NAME !

Let's stop here and create a new bug about 'Fixing duplicated HDU_NAME / Image identity checks' ...

**Comment by buthanoid** [comment]
I agree that the file was corrupted and it was coming from a previous WISARD run but still:
- why, because I have selected an input hdu named "PI_GRU_X", when I run, should I refuse any result HDU named "PI_GRU_X" ? 
The HDU name is used to identify a HDU inside an oiFitsFile (by using an alias instead of the HDU index). Thus I think we should not use it to compare HDUs coming from different files. The information about the content of the HDU is insufficient and the algorithms tend to reuse names.

**Comment by bourgesl** [comment]
Well, we should read again the standard. For results, the first hdu is the result, whatever hdu_name it has...
Let's plug another rule for results: always add first hdu (do not check selectedImage) ?

**Comment by buthanoid** [comment]
I added your solution : not checking first HDU of a serviceResult and checking the other HDUs

**Comment by bourgesl** [comment]
Probably it is easier to fix addFitsImageHDUs() to handle your particular case: alwaysAddFirst, and it avoids code duplication

File: src/main/java/fr/jmmc/oimaging/model/IRModel.java [hunk 1]
```diff
@@ -248,29 +248,56 @@ public void loadOifitsFile(final OIFitsFile userOifitsFile) {
...
 
         final boolean added = !addedHdus.isEmpty();
 
         // avoid modifying input file while loading OIFits (reentrance):
         if (hdus != oifitsFile.getFitsImageHDUs()) {
             final FitsImageHDU selectedInitImage;
             if (added) {
-                // select first added hdu as selected input
-                selectedInitImage = addedHdus.get(0);
+                if (addAndSelectFirstHDU && hdus.get(0) != addedHdus.get(0)) {
```
Comment (bourgesl) [inline src/main/java/fr/jmmc/oimaging/model/IRModel.java:292]: looks tricky: what should be the selected HDU ?

File: src/main/java/fr/jmmc/oimaging/model/IRModel.java [hunk 2]
```diff
@@ -248,29 +248,56 @@ public void loadOifitsFile(final OIFitsFile userOifitsFile) {
      * @return true if some hdu have been added
      */
     public boolean addFitsImageHDUs(final List<FitsImageHDU> hdus, final String filename,
-                                    final Set<String> inputImageRefs) {
+            final Set<String> inputImageRefs) {
```
Comment (bourgesl) [inline src/main/java/fr/jmmc/oimaging/model/IRModel.java:251]: delete this method to force calling the addFitsImageHDUs with appropriate flags (inputImageRefs, addAndSelectFirstHDU)

**Comment by bourgesl** [comment]
Good to go

**Pull request merged**